File: 010 Bubble Sort/Bubble-Sort.py
# Best - O(n) Time | O(1) Space
# Average - O(n^2) Time | O(1) Space
# Worst - O(n^2) Time | O(1) Sace
def bubbleSort(array):
	isSorted = False
	counter = 0 
	while not isSorted:
		isSorted = True
		for i in range(len(array) - 1 - counter):
			if array[i] > array[i + 1]:
				swap(i, i + 1, array)
				isSorted = False
		counter += 1
	return array
def swap(x,y,arr):
	arr[x], arr[y] = arr[y], arr[x]

File: 010 Bubble Sort/Bubble-Sort.js
function bubbleSort(array) {
  let isSorted = false;
	let counter = 0;
	while (!isSorted) {
		isSorted = true;
		for (let i = 0; i < array.length - 1 - counter; i++) {
			if (array[i] > array[i + 1])
				swap(i, i + 1, array)
				isSorted = false
		}
		counter ++;
	}
	return array;
}

function swap(a,b, array) {
	[array[a], array[b]] = [array[b], array[a]];
}